runtime/interpreter/interpreter.go

```diff
@@ -3563,6 +3569,61 @@ func (interpreter *Interpreter) authAccountLinkFunction(addressValue AddressValu
 	)
 }
 
+var authAccountReferenceStaticType = ReferenceStaticType{
+	BorrowedType:   PrimitiveStaticTypeAuthAccount,
+	ReferencedType: PrimitiveStaticTypeAuthAccount,
+}
+
+func (interpreter *Interpreter) authAccountLinkAccountFunction(addressValue AddressValue) *HostFunctionValue {
+
+	// Converted addresses can be cached and don't have to be recomputed on each function invocation
+	address := addressValue.ToAddress()
+
+	return NewHostFunctionValue(
+		interpreter,
+		func(invocation Invocation) Value {
+			interpreter := invocation.Interpreter
+
+			newCapabilityPath, ok := invocation.Arguments[0].(PathValue)
+			if !ok {
+				panic(errors.NewUnreachableError())
+			}
+
+			newCapabilityDomain := newCapabilityPath.Domain.Identifier()
+			newCapabilityIdentifier := newCapabilityPath.Identifier
+
+			if interpreter.storedValueExists(
+				address,
+				newCapabilityDomain,
+				newCapabilityIdentifier,
+			) {
+				return Nil
+			}
+
+			accountLinkValue := NewAccountLinkValue(interpreter)
+
+			interpreter.WriteStored(
+				address,
+				newCapabilityDomain,
+				newCapabilityIdentifier,
+				accountLinkValue,
+			)
+
+			return NewSomeValueNonCopying(
+				interpreter,
+				NewStorageCapabilityValue(
```

Comment:
Are 'account capabilities' also represented using `StorageCapabilityValue`s?

Reply:
Correct. 

When linking to a path with the `AuthAccount.link` function, an `interpreter.PathLink` (formerly `Link`) is stored in storage.
When linking to an account with the new `AuthAccount.linkAccount` function, an `interpreter.AccountLink` is stored in the account.

In both cases, when acquiring a capability, e.g. using `getCapability`, a `StorageCapabilityValue` is returned. This is because in both cases, we are looking up a path in an account. Depending on what is stored in the path, `PathLink` or `AccountLink`, we return a respective reference value, a `StorageReferenceValue` for `PathLink` (after following the links to the final target), or an `AccountReferenceValue` for an `AcountLink`.

Again, in both cases for `StorageReferenceValue` and `AccountReferenceValue`, for each use, e.g. member access, we dereference/check that the link still exists after the capability was borrowed.

Reply:
Added this as a comment in [3f45be7](https://github.com/onflow/cadence/pull/2160/commits/3f45be723303a6ed19324d8b25064be018941c11)